proxy/src/session.rs

```diff
@@ -140,6 +140,17 @@ pub fn set_settings(store: &kv::Store, settings: settings::Settings) -> Result<(
     set(store, KEY_CURRENT, sess)
 }
 
+/// Gets the [`settings::Settings`] in the current session.
+///
+/// # Errors
+///
+/// Errors if access to the session state fails.
+pub fn get_settings(store: &kv::Store) -> Result<settings::Settings, error::Error> {
```

Comment:
This will produce an incorrect session, check the existing one, which we should use for now:

https://github.com/radicle-dev/radicle-upstream/blob/0c3e457757988d0a05feab43727421c709f06778/proxy/src/session.rs#L59-L97

Reply:
This requires the registry to be passed in though?

Reply:
For now that's the case yes, if we want a consistent view on the session that is what it takes. We have some work to do to streamline the object graph necessary to full-fill our requests.